doc/overviews/inference.md

```diff
@@ -0,0 +1,412 @@
+
+Example with 3 PromptGuardPolicy and 3 HTTPRoutes, without merges nor overrides in place:
+
+- PromptGuardPolicy A → HTTPRoute A (`a.toystore.com`)
+- PromptGuardPolicy B → HTTPRoute B (`b.toystore.com`)
+- PromptGuardPolicy W → HTTPRoute W (`*.toystore.com`)
+
+Expected behavior:
+
+- Request to `a.toystore.com` → PromptGuardPolicy A will be enforced
+- Request to `b.toystore.com` → PromptGuardPolicy B will be enforced
+- Request to `other.toystore.com` → PromptGuardPolicy W will be enforced
+
+### `when` conditions
+
+`when` conditions can be used to scope a PromptGuardPolicy or TokenRateLimitPolicy (i.e. to filter the traffic to which a policy or policy rule applies) without any coupling to the underlying network topology.
+
+Use `when` conditions to conditionally activate policies and policy rules based on attributes that cannot be expressed in the HTTPRoutes' `spec.hostnames` and `spec.rules.matches` fields, or in general in PromptGuardPolicies or TokenRateLimitPolicies that target a Gateway.
+
+The selectors within the `when` conditions of an PromptGuardPolicy or TokenRateLimitPolicy are a subset of Kuadrant's Well-known Attributes ([RFC 0002](https://github.com/Kuadrant/architecture/blob/main/rfcs/0002-well-known-attributes.md)). Check out the reference for the full list of supported selectors.
+
+### Known limitations
+
+- PromptGuardPolicy & TokenRateLimitPolicy can only target HTTPRoutes/Gateways defined within the same namespace of the them.
+
+## Implementation details
+
+### Inference service
+
+The Inference service can run in 2 different modes: Single tenant and multi tenant.
+
+#### Single tenant mode
+
+In single tenant mode, the service can be used without the Kuadrant operator.
+It depends on Envoy Proxy, without Gateway API or a Gateway API provider.
+However, the biggest limitation with this mode is that the configuration is static.
+For example, you configure the service with the prompt guarding LLM and risk categories once on startup.
+All requests that are checked by the service are subject to the same prompt guarding configuration.
+
+Here is a request flow diagram for the single tenant architecture:
+
+![ai_service_single_tenant](./Arch%20Overview%20v1%20-%20ai_service_single_tenant.jpg)
+
+In this setup, Envoy Proxy is configured manually with the service as an 'ExternalProcessor' (ext_proc) filter.
+The service is configured as a 'cluster' in Enovy Proxy.
+The service listens on a different grpc port for each feature (like prompt guarding or semantic caching).
+This allows for different features to be executed at different stages of the request lifecycle.
```

Comment:
I question does this section belong here? Not saying the information is not valuable, but the was an effort to remove reference and guides of the sub components. This seems to be introducing a reference to sub component. I know at the moment there is no inference service repo, that is where the documents around running it in a stand alone manner should live.

Reply:
I think the words here are sufficiently abstracted and at a high level that it's fine to have here.
Any further details on how to run the component in that way should definitely live in that component.